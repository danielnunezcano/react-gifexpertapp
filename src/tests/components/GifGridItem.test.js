import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Test GifGridItem.js", () => {
  const data = { id: 1, title: "Un título", url: "https://localhost/algo.jpg" };

  let wrapper = shallow(<GifGridItem {...data} />);
  test("Debe de mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe tener un párrafo con el titulo", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(data.title);
  });
  test("debe de tener la imagen igual al url y alt de los props", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(data.url);
    expect(img.prop("alt")).toBe(data.title);
  });
  test("debe de tener el animate__fadeIn", () => {
    const img = wrapper.find("div");
    const className = img.props("class").className;
    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
