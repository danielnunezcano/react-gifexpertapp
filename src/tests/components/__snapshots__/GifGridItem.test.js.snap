// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifGridItem.js Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    key="1"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
